src/middlewared/middlewared/alembic/versions/12.0/2021-12-22_12-59_add_server_tls_uri_to_s3_config.py

```diff
@@ -7,13 +7,29 @@
 import sqlalchemy as sa
 from OpenSSL import crypto
 
+from middlewared.validators import Hostname
+
 # revision identifiers, used by Alembic.
 revision = '9c11f6c6f152'
 down_revision = 'fee786dfe121'
 branch_labels = None
 depends_on = None
 
 
+def is_valid_hostname(hostname: str):
+    """
+    Validates hostname and makes sure it
+    does not contain a valid card.
+    """
+    validate_hostname = Hostname()
+    try:
+        validate_hostname(hostname)
+    except ValueError:
+        return False
+    else:
+        return "*" not in hostname
```

Comment:
We should update the `Hostname` validator so that it correctly validates an actual hostname and have this there instead